lib/db/mysql.js

```diff
@@ -112,31 +124,33 @@ module.exports = function (log, error) {
 
   // this will be called from outside this file
   MySql.connect = function(options) {
-    // check that the database patch level is what we expect (or one above)
-    var mysql = new MySql(options)
-
-    // Select : dbMetadata
-    // Fields : value
-    // Where  : name = $1
-    var DB_METADATA = 'CALL dbMetadata_1(?)'
-
-    return mysql.readFirstResult(DB_METADATA, [options.patchKey])
-      .then(
-        function (result) {
-          mysql.patchLevel = +result.value
-
-          log.info('connect', {
-            patchLevel: mysql.patchLevel,
-            patchLevelRequired: patch.level
-          })
+    return new P.resolve().then(() => {
```

Comment:
The use of `new P.resolve()` here seems odd; is it necessary at all, and if so should we remove the `new`?